bundles/org.openhab.core.config.discovery.upnp/src/main/java/org/openhab/core/config/discovery/upnp/UpnpDiscoveryParticipant.java

```diff
@@ -62,4 +62,21 @@
      */
     @Nullable
     ThingUID getThingUID(RemoteDevice device);
+
+    /**
+     * The JUPnP library strictly follows the UPnP specification insofar as if a device fails to send its next
+     * 'ssdp:alive' notification within its declared 'maxAge' period, it is immediately considered to be gone. But
+     * unfortunately some types of OpenHAB bindings have devices that can sometimes be a bit late in sending their
+     * 'ssdp:alive' notifications even though they have not really gone offline, which means that such devices are
+     * repeatedly removed from, and (re)added to, the InBox.
+     *
+     * To prevent this, a binding that implements a UpnpDiscoveryParticipant may OPTIONALLY implement this method to
+     * specify an additional delay period (grace period) to wait before the device is removed from the Inbox.
+     *
+     * @param device the upnp device on the network
+     * @return the additional grace period delay in seconds before the device will be removed from the Inbox
+     */
+    default int getRemovalGracePeriodSeconds(RemoteDevice device) {
```

Comment:
Two questions:

1. Why do we need the parameter `RemoteDevice`? You are neither using it here nor in https://github.com/openhab/openhab-addons/pull/9985.

2. Shouldn't we return a `long` value instead of an `int`? The `delay` parameter in `schedule(...)` is a `long` too.

Reply:
@kaikreuzer thanks for your review. Concerning your two points..

1. Regarding the extra argument: I thought that there could be cases in future where the discovery service might be handling more than one class of device; so it would allow the binding to have different gracePeriods for different device types. But the Hue discovery only handles the Hue Bridge so the argument is not used..
2. Re long vs int: ok I will fix that.

Reply:
I am not @kaikreuzer , but I take it as a compliment. 😉

Reply:
>I am not @kaikreuzer , but I take it as a compliment

Sorry for that :( 
But I edited the post to show your handle now :)